### PR TITLE
Support referencing responses

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js [hunk 1]
```diff
@@ -48,9 +49,10 @@ const parseComponentMember = R.curry((context, parser, member) => {
   // Create a Member Element with `member.key` as the key
   const Member = R.constructN(2, context.namespace.elements.Member)(member.key);
 
-  const parseResult = pipeParseResult(context.namespace,
-    parser(context),
-    Member)(member.value);
+  const parseResult = R.map(
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js]: NOTE, this minor refactoring of this code is required to support referencing a component that has multiple values. For example a "response object" in OAS 3 maps to multiple HttpResponse elements. The map is simpler than previous implementation where we are mapping (wrapping) all non annotation elements in members.

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseResponsesObject.js [hunk 2]
```diff
@@ -31,8 +34,26 @@ const isResponseField = R.anyPass([isStatusCode, isStatusCodeRange, hasKey('defa
 function parseResponsesObject(context, element) {
   const { namespace } = context;
 
+  const createUnsupportedStatusCodeWarning = (member) => {
+    let message;
+
+    if (member.key.toValue() === 'default') {
+      message = `'${name}' default responses unsupported`;
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseResponsesObject.js]:
```suggestion
      message = `'${name}' default responses are unsupported`;
```

File: packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js [hunk 3]
```diff
@@ -12,38 +12,19 @@ describe('Response Object', () => {
...
-  describe('status code', () => {
-    it('provides a warning for default responses', () => {
-      const response = new namespace.elements.Member('default', {});
-      const result = parse(context, response);
-
-      expect(result.length).to.equal(1);
-      expect(result).to.contain.warning("'Response Object' default responses unsupported");
-    });
-
-    it('provides a warning for range responses', () => {
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js]: This test was removed, but I don't see a replacement

File: packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js [hunk 4]
```diff
@@ -12,38 +12,19 @@ describe('Response Object', () => {
...
     });
     const result = parse(context, response);
 
     expect(result.length).to.equal(1);
     const httpResponse = result.get(0);
     expect(httpResponse).to.be.instanceof(namespace.elements.HttpResponse);
-    expect(httpResponse.attributes.get('statusCode').toValue()).to.be.equal('200');
-  });
-
-  describe('status code', () => {
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js]: This test was removed, but I don't see a replacement

Comment (kylef) [reply packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js]: There is already a test for this behaviour in parseResponsesObject-test.js, you can see it in the diff below.

File: packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js (hunk 3 above)
Comment (kylef) [reply packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseResponseObject-test.js]: There is already a test for this behaviour in parseResponsesObject-test.js, you can see it in the diff below.

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseResponsesObject.js (hunk 2 above)
Comment (kylef) [reply packages/fury-adapter-oas3-parser/lib/parser/oas/parseResponsesObject.js]: Added in 41cbb86

**Pull request merged**